jbmc/regression/jbmc-strings/CompareToConstantEvaluation/Main.java

```diff
@@ -0,0 +1,14 @@
+public class Main {
+    public void constantCompareTo() {
```

Comment:
These tests should really also consider cases that *shouldn't* be simplified, such as "ab".endsWith("abc"). Try to include a case for each `if (...) return true` below.

Reply:
Not sure I understand. The example your mention *should* actually  be simplified, though the assertions should fail. I'm currently writing tests for failing assertions.